server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -152,9 +154,6 @@ public IndexRequest(String index, String type, String id) {
     @Override
     public ActionRequestValidationException validate() {
         ActionRequestValidationException validationException = super.validate();
-        if (type == null) {
-            validationException = addValidationError("type is missing", validationException);
-        }
```

Comment:
Any reason this is removed here but still validated in DeleteRequest?

Reply:
No, I can't think why they should differ